### ISSUE TITLE
Synchronous compute methods should be immediately invoked

### ISSUE THREAD

**Issue by darrenburns** [Textualize/textual#1218]
The test below fails because the compute method is not called immediately.
This will catch out users and lead to subtle issues because there's no telling when a computed value will be available.
Synchronous compute methods should be invoked immediately rather than queued, so that the user can safely make use of a computed value after updating a reactive, without worrying about whether it has been recomputed or not.

```python
async def test_superpower_method_call_order():
    class CallOrder(App):
        count = reactive(OLD_VALUE, init=False)
        count_times_ten = reactive(OLD_VALUE * 10)
        calls = []

        def validate_count(self, value: int) -> int:
            self.calls.append(f"validate {value}")
            return value + 1

        def watch_count(self, value: int) -> None:
            self.calls.append(f"watch {value}")

        def compute_count_times_ten(self) -> int:
            self.calls.append(f"compute {self.count}")
            return self.count * 10

    app = CallOrder()
    async with app.run_test():
        app.count = NEW_VALUE
        assert app.calls == [
            # The validator receives NEW_VALUE, since that's what the user
            # set the reactive attribute to...
            f"validate {NEW_VALUE}",
            # The validator adds 1 to the new value, and this is what should
            # be passed into the watcher...
            f"watch {NEW_VALUE + 1}",
            # The compute method accesses the reactive value directly, which
            # should have been updated by the validator to NEW_VALUE + 1.
            f"compute {NEW_VALUE + 1}",
        ]
```

**Comment by willmcgugan** [comment]
I think this is the case now. We need to confirm. Close if it is already working, or fix if not.

**Comment by darrenburns** [comment]
There's an xfail test, so if it started working then [this](https://github.com/Textualize/textual/blob/e968ca0e50faf3b9255fd279a3b19836c49f9255/tests/test_reactive.py#L207) test should have started failing (unless the test is wrong).

**Comment by rodrigogiraoserrao** [comment]
Closed by #1558
